contracts/base/escrow/dapps/Aave.sol

```diff
@@ -0,0 +1,127 @@
+        IERC20(tokenAddress).safeApprove(address(aaveLendingPool), amount);
+        aaveLendingPool.deposit(tokenAddress, amount, address(this), 0);
+
+        uint256 aTokenBalanceAfterDeposit = aToken.balanceOf(address(this));
+        require(
+            aTokenBalanceAfterDeposit > aTokenBalanceBeforeDeposit,
+            "AAVE_BALANCE_NOT_INCREASED"
+        );
+
+        _tokenUpdated(address(aToken));
+        _tokenUpdated(tokenAddress);
+
+        uint256 tokenBalanceAfterDeposit = _balanceOf(tokenAddress);
+        emit AaveDeposited(
+            tokenAddress,
+            address(aToken),
+            amount,
+            tokenBalanceAfterDeposit,
+            aTokenBalanceAfterDeposit
+        );
+    }
+
+    /**
+        @notice This function withdraws the user's aTokens for a specific amount 
+        @param tokenAddress address of the token
+        @param amount amount of the underlying tokens to withdraw
+     */
+    function withdraw(address tokenAddress, uint256 amount) public onlyOwner() {
+        IAToken aToken = _getAToken(tokenAddress);
+        IAaveLendingPool aaveLendingPool = _getAavePoolAddress(tokenAddress);
+        uint256 aTokenBalanceBeforeWithdraw = aToken.balanceOf(address(this));
+        require(aTokenBalanceBeforeWithdraw >= 0, "NO_BALANCE_TO_WITHDRAW");
+        uint256 result = aaveLendingPool.withdraw(tokenAddress, amount, address(this));
+        uint256 aTokenBalanceAfterWithdraw = aToken.balanceOf(address(this));
+        require(
+            aTokenBalanceAfterWithdraw < aTokenBalanceBeforeWithdraw,
+            "AAVE_WITHDRAWAL_ERROR"
+        );
+
+        _tokenUpdated(address(aToken));
+        _tokenUpdated(tokenAddress);
+
+        uint256 tokenBalanceAfterWithdrawal = _balanceOf(tokenAddress);
+        emit AaveWithdrawn(
+            tokenAddress,
+            address(aToken),
+            amount,
+            tokenBalanceAfterWithdrawal,
+            aTokenBalanceAfterWithdraw
+        );
+    }
+
+    /**
+        @notice Grabs the Aave lending pool instance from the Aave lending pool address provider
+        @param tokenAddress The token address to get the lending pool for
+        @return IAaveLendingPool instance address
+     */
+    function _getAavePoolAddress(address tokenAddress)
+        internal
+        view
+        returns (IAaveLendingPool)
+    {
+        return
+            IAaveLendingPool(
+                IAaveLendingPoolAddressesProvider(tokenAddress).getLendingPool()
+            );
+    }
+
+    /**
+        @notice Grabs the aToken instance from the lending pool
+        @param tokenAddress The underlying asset address to get the aToken for
+        @return IAToken instance
+     */
+    function _getAToken(address tokenAddress) internal view returns (IAToken) {
+        return
+            IAToken(
+                _getAavePoolAddress(tokenAddress)
+                    .getReserveData(tokenAddress)
```

Comment:
Is it correct to use tokenAddress in both calls?

Reply:
Yes: 

The first _getAavePoolAddress,  grabs the lending pool instance from the Aave lending pool address provider. 
https://github.com/teller-protocol/teller-protocol-v1/pull/252/files#diff-febb5fbb878382a74affe256f1bea611efac861dccff043b062b658463ae1a53R99

The second getReserveData gets the reserve data for the retrieved lending pool from the loaded interface